src/Spago/Command/Publish.purs

```diff
@@ -476,5 +481,32 @@ locationIsInGitRemotes location = do
           Location.Git { url } -> r.url == url
           Location.GitHub { owner, repo } -> r.owner == owner && r.repo == repo
 
+inferLocationAndWriteToConfig :: ∀ a. WorkspacePackage -> Spago (PublishEnv a) Boolean
+inferLocationAndWriteToConfig selectedPackage = do
+  Git.getRemotes Nothing >>= case _ of
+    Left err ->
+      die $ toDoc err
+    Right remotes ->
+      case Array.find (_.name >>> eq "origin") remotes of
+        Nothing ->
+          pure false
+        Just origin -> do
+          let
+            subdir
+              | Config.isRootPackage selectedPackage = Nothing
+              | otherwise = Just $ withForwardSlashes selectedPackage.path
+            configPath
+              | Config.isRootPackage selectedPackage = "spago.yaml"
+              | otherwise = Path.concat [ selectedPackage.path, "spago.yaml" ]
+            location
+              | String.contains (String.Pattern "github.com") origin.url =
+                  Location.GitHub { owner: origin.owner, repo: origin.repo, subdir }
+              | otherwise =
+                  Location.Git { url: origin.url, subdir }
```

Comment:
All of this is "correct" in the sense that eventually we're going to support it in this way - however right now the registry only supports `GitHub` locations, with `subdir: Nothing` (for backwards compatibility with Bower and Pursuit). So I'd like to keep this code, but error out in unsupported cases (and record that in the fixtures too, so they will change once we start supporting this)

Reply:
Oh, I didn't realize that. Does the publish command already error out if you specify a subdir? Or would you like me to add that too?

Reply:
Done